cmd/spiffe-helper/config/config.go

```diff
@@ -15,28 +15,35 @@ import (
 	"github.com/spiffe/spiffe-helper/pkg/sidecar"
 )
 
+const (
+	DaemonModeFlagName = "daemon-mode"
+)
+
 const (
 	defaultAgentAddress      = "/tmp/spire-agent/public/api.sock"
 	defaultCertFileMode      = 0644
 	defaultKeyFileMode       = 0600
 	defaultJWTBundleFileMode = 0600
 	defaultJWTSVIDFileMode   = 0600
+	defaultHealthCheckPort   = 8081
+	defaultHealthCheckPath   = "/healthz"
 )
 
 type Config struct {
-	AddIntermediatesToBundle bool   `hcl:"add_intermediates_to_bundle"`
-	AgentAddress             string `hcl:"agent_address"`
-	Cmd                      string `hcl:"cmd"`
-	CmdArgs                  string `hcl:"cmd_args"`
-	PIDFileName              string `hcl:"pid_file_name"`
-	CertDir                  string `hcl:"cert_dir"`
-	CertFileMode             int    `hcl:"cert_file_mode"`
-	KeyFileMode              int    `hcl:"key_file_mode"`
-	JWTBundleFileMode        int    `hcl:"jwt_bundle_file_mode"`
-	JWTSVIDFileMode          int    `hcl:"jwt_svid_file_mode"`
-	IncludeFederatedDomains  bool   `hcl:"include_federated_domains"`
-	RenewSignal              string `hcl:"renew_signal"`
-	DaemonMode               *bool  `hcl:"daemon_mode"`
+	AddIntermediatesToBundle bool               `hcl:"add_intermediates_to_bundle"`
+	AgentAddress             string             `hcl:"agent_address"`
+	Cmd                      string             `hcl:"cmd"`
+	CmdArgs                  string             `hcl:"cmd_args"`
+	PIDFileName              string             `hcl:"pid_file_name"`
+	CertDir                  string             `hcl:"cert_dir"`
+	CertFileMode             int                `hcl:"cert_file_mode"`
+	KeyFileMode              int                `hcl:"key_file_mode"`
+	JWTBundleFileMode        int                `hcl:"jwt_bundle_file_mode"`
+	JWTSVIDFileMode          int                `hcl:"jwt_svid_file_mode"`
+	IncludeFederatedDomains  bool               `hcl:"include_federated_domains"`
+	RenewSignal              string             `hcl:"renew_signal"`
+	DaemonMode               *bool              `hcl:"daemon_mode"`
+	HealthCheck              *HealthCheckConfig `hcl:"health_checks"`
```

Comment:
I think you can just make this a regular struct, not pointer.

Reply:
I thought it needed to be a pointer for backwards compatibility with existing configs that don't have this section?

Reply:
It depends. If we want it to default to on then having it be a pointer makes sense. You can check for nil and set the default values and enable it. Thats why the `DaemonMode` config is a pointer as we want it to default to daemonMode if nothing is specified.

In this case we probably want the health check to be disabled by default so a regular non pointer will default to zero values which will have the health checks disabled.

Reply:
OK